File: server/backup.mjs
import { DatabaseSync, backup } from 'node:sqlite';
import { existsSync, mkdirSync, chmodSync } from 'node:fs';
import { resolve, dirname } from 'node:path';
const database=resolve(process.env.DATA_DIR||'data','returns.sqlite');
if(!existsSync(database))throw new Error('Datenbank nicht gefunden.');
const destination=resolve(process.argv[2]||`data/backups/returns-${new Date().toISOString().replaceAll(':','-')}.sqlite`);
if(existsSync(destination))throw new Error('Zieldatei existiert bereits. Bitte einen neuen Dateinamen wählen.');
mkdirSync(dirname(destination),{recursive:true,mode:0o700});
const db=new DatabaseSync(database,{readOnly:true});
try{await backup(db,destination);chmodSync(destination,0o600);console.log('Datenbanksicherung erstellt.');}finally{db.close();}

File: server/backup.py
import os
import sys
from datetime import datetime, timezone
from pathlib import Path
from dotenv import load_dotenv
from .database import backup_database


def main():
    load_dotenv()
    data = Path(os.environ.get('DATA_DIR', 'data'))
    stamp = datetime.now(timezone.utc).strftime('%Y-%m-%dT%H-%M-%S-%fZ')
    destination = Path(sys.argv[1]) if len(sys.argv) > 1 else data / 'backups' / f'returns-{stamp}.sqlite'
    backup_database(data / 'returns.sqlite', destination)
    print('Datenbanksicherung erstellt.')


if __name__ == '__main__':
    main()
